packages/react/src/components/DateTimePicker/DateTimePickerV2.test.jsx

```diff
@@ -528,47 +1047,133 @@ describe('DateTimePickerV2', () => {
+          },
+        }}
+      />
+    );
+
+    expect(screen.getByText('2021-08-01 12:34 to 2021-08-06 10:49')).toBeVisible();
+  });
+
+  it('should parse default values with start/end (new time spinner)', () => {
+    render(
+      <DateTimePicker
+        {...dateTimePickerProps}
+        useNewTimeSpinner
+        hasTimeInput
+        defaultValue={{
+          timeRangeKind: PICKER_KINDS.ABSOLUTE,
+          timeRangeValue: {
+            start: '2021-08-01 12:34',
+            end: '2021-08-06 10:49',
+          },
+        }}
+      />
+    );
+
+    expect(screen.getByText('2021-08-01 12:34 to 2021-08-06 10:49')).toBeVisible();
+  });
+
+  it('should not render a time picker if hasTimeInput:false', () => {
+    render(
+      <DateTimePicker
+        {...dateTimePickerProps}
+        id="picker-test"
+        hasTimeInput={false}
+        defaultValue={{
+          timeRangeKind: PICKER_KINDS.ABSOLUTE,
+          timeRangeValue: {
+            start: new Date(2021, 7, 1, 12, 34, 0),
+            end: new Date(2021, 7, 6, 10, 49, 0),
+          },
+        }}
+      />
+    );
+
+    userEvent.click(screen.getByText('2021-08-01 12:34 to 2021-08-06 10:49'));
+    expect(screen.queryByLabelText('Start time')).toBeNull();
+    expect(screen.queryByLabelText('End time')).toBeNull();
+  });
+
+  it('should not render a time picker if hasTimeInput:false (new time spinner)', () => {
+    render(
+      <DateTimePicker
+        {...dateTimePickerProps}
+        useNewTimeSpinner
+        id="picker-test"
+        hasTimeInput={false}
+        defaultValue={{
+          timeRangeKind: PICKER_KINDS.ABSOLUTE,
+          timeRangeValue: {
+            start: new Date(2021, 7, 1, 12, 34, 0),
+            end: new Date(2021, 7, 6, 10, 49, 0),
+          },
+        }}
+      />
+    );
+
+    userEvent.click(screen.getByText('2021-08-01 00:00 to 2021-08-06 00:00'));
+    expect(screen.queryByLabelText('Start time')).toBeNull();
+    expect(screen.queryByLabelText('End time')).toBeNull();
+  });
+
+  it('should fallback to minute intervals if none given in a relative defaultValue', () => {
     render(
       <DateTimePicker
         {...dateTimePickerProps}
-        hasTimeInput
+        id="picker-test"
+        hasTimeInput={false}
         defaultValue={{
-          timeRangeKind: PICKER_KINDS.ABSOLUTE,
+          timeRangeKind: PICKER_KINDS.RELATIVE,
           timeRangeValue: {
-            start: '2021-08-01 12:34',
-            end: '2021-08-06 10:49',
+            lastNumber: 20,
+            relativeToWhen: RELATIVE_VALUES.TODAY,
+            relativeToTime: '13:30',
           },
         }}
       />
     );
+    jest.runAllTimers();
 
-    expect(screen.getByText('2021-08-01 12:34 to 2021-08-06 10:49')).toBeVisible();
+    userEvent.click(screen.getByTestId('date-time-picker__field'));
+    expect(screen.getByText('minutes')).toBeInTheDocument();
   });
 
-  it('should not render a time picker if hasTimeInput:false', () => {
```

Comment:
There still seems to be changed test here. These should stay to ensure the old functionality is being preserved, no?

Reply:
> There still seems to be changed test here. These should stay to ensure the old functionality is being preserved, no?

The test was moved to line 1087.